Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/Instance.cpp

```diff
@@ -28,24 +29,71 @@ namespace AzToolsFramework
         }
 
         Instance::Instance(AZStd::unique_ptr<AZ::Entity> containerEntity)
+            : m_parent(nullptr)
+            , m_alias(GenerateInstanceAlias())
+            , m_instanceEntityMapper(AZ::Interface<InstanceEntityMapperInterface>::Get())
+            , m_templateInstanceMapper(AZ::Interface<TemplateInstanceMapperInterface>::Get())
         {
-            m_instanceEntityMapper = AZ::Interface<InstanceEntityMapperInterface>::Get();
-
             AZ_Assert(m_instanceEntityMapper,
                 "Instance Entity Mapper Interface could not be found. "
                 "It is a requirement for the Prefab Instance class. "
                 "Check that it is being correctly initialized.");
 
-            m_templateInstanceMapper = AZ::Interface<TemplateInstanceMapperInterface>::Get();
+            AZ_Assert(m_templateInstanceMapper,
+                "Template Instance Mapper Interface could not be found. "
+                "It is a requirement for the Prefab Instance class. "
+                "Check that it is being correctly initialized.");
+
+            if (containerEntity)
+            {
+                m_containerEntity = AZStd::move(containerEntity);
+            }
+            else
+            {
+                AliasPath absoluteInstancePath;
+                absoluteInstancePath.Append(m_alias);
+                absoluteInstancePath.Append(PrefabDomUtils::ContainerEntityName);
+
+                AZ::EntityId containerEntityId = InstanceEntityIdMapper::GenerateEntityIdForAliasPath(absoluteInstancePath);
+                m_containerEntity = AZStd::make_unique<AZ::Entity>(containerEntityId);
```

Comment:
repeated: If there is no parent instance provided, then GenerateEntityIdForAliasPath() doesn't add much value because after adding a parent instance, the id changes because of we generate ids based on hashes of absolute paths. So, can we just skip this part and create an entity with a random entity id?